ahnlich/ai/src/tests/aiproxy_test.rs

```diff
@@ -446,3 +457,129 @@ async fn test_ai_proxy_fails_db_server_unavailable() {
+            model: AIModel::Llama3,
+            predicates: HashSet::from_iter([]),
+            non_linear_indices: HashSet::new(),
+        },
+        AIQuery::CreateStore {
+            r#type: AIStoreType::Binary,
+            store: store_name_2.clone(),
+            model: AIModel::Llama3,
+            predicates: HashSet::from_iter([]),
+            non_linear_indices: HashSet::new(),
+        },
+        AIQuery::DropStore {
+            store: store_name,
+            error_if_not_exists: true,
+        },
+    ]);
+
+    let mut expected = AIServerResult::with_capacity(3);
+
+    expected.push(Ok(AIServerResponse::Unit));
+    expected.push(Ok(AIServerResponse::Unit));
+    expected.push(Ok(AIServerResponse::Del(1)));
+
+    let mut reader = BufReader::new(first_stream);
+    query_server_assert_result(&mut reader, message, expected).await;
+
+    // write flag should show that a write has occured
+    assert!(write_flag.load(Ordering::SeqCst));
+    // Allow some time for persistence to kick in
+    tokio::time::sleep(Duration::from_millis(200)).await;
+    // start another server with existing persistence
+
+    let persisted_server = AIProxyServer::new(&AI_CONFIG_WITH_PERSISTENCE)
+        .await
+        .unwrap();
+
+    // Allow some time for the server to start
+    tokio::time::sleep(Duration::from_millis(100)).await;
+
+    let address = persisted_server
+        .local_addr()
+        .expect("Could not get local addr");
+    let write_flag = persisted_server.write_flag();
+    let _ = tokio::spawn(async move { persisted_server.start().await });
+    let second_stream = TcpStream::connect(address).await.unwrap();
+    let mut reader = BufReader::new(second_stream);
+
+    let message = AIServerQuery::from_queries(&[AIQuery::ListStores]);
+
+    let mut expected = AIServerResult::with_capacity(1);
+
+    expected.push(Ok(AIServerResponse::StoreList(HashSet::from_iter([
+        AIStoreInfo {
+            name: store_name_2.clone(),
+            r#type: AIStoreType::Binary,
+            model: AIModel::Llama3,
+            embedding_size: AIModel::Llama3.embedding_size().into(),
+        },
+    ]))));
+
+    query_server_assert_result(&mut reader, message, expected).await;
+    assert!(!write_flag.load(Ordering::SeqCst));
+    // delete persistence
+    let _ = std::fs::remove_file(&*PERSISTENCE_FILE);
+}
+
+#[tokio::test]
+async fn test_ai_proxy_destroy_database() {
+    let address = provision_test_servers().await;
+    let second_stream = TcpStream::connect(address).await.unwrap();
+    let store_name = StoreName(String::from("Deven Kicks"));
+    let message = AIServerQuery::from_queries(&[
+        AIQuery::CreateStore {
+            r#type: AIStoreType::RawString,
+            store: store_name.clone(),
+            model: AIModel::Llama3,
+            predicates: HashSet::from_iter([]),
+            non_linear_indices: HashSet::new(),
+        },
+        AIQuery::ListStores,
+        AIQuery::DestoryDatabase,
```

Comment:
More like destroy store or something

Reply:
destroy stores ?
It's a db wide action

Reply:
Yes but we can't refer to the ai store as a database.... might also falsely give the impression that we are destroying the actual database i.e the vector db